docs/err.3
.TH ERR 3 "25 July 1990" local "Utility routines"
.SH NAME 
warning, error, fatal \- consistent error handling
.SH SYNOPSIS
#include <setjmp.h>
.br
#include <err.h>
.LP
warning( fmt, ... )
.br
char *fmt;
.LP
error( fmt, ... )
.br
char *fmt;
.LP
fatal( fmt, ... )
.br
char *fmt;
.LP
fatal_nr( nr, fmt, ... )
.br
int	nr;
.br
char *fmt;
.LP
extern int err_nr_errors, err_nr_warnings;
.br
extern int *err_warning_jmpbuf, *err_error_jmpbuf, *err_fatal_jmpbuf;
.SH DESCRIPTION
These routines provide a way to handle all kind of errors in a consistent way.
The differences between the routines are:
.I warning()
is intended to be used when the error encountered is to be reported, but
will probably not harm the result.
.I error()
should be called when the error will harm the result (it will probably not be 
valid), but execution can continue;
.I fatal()
should be called when the error means that the program can not continue.
.I fatal()
will exit with a return value of 1;
.I fatal_nr()
will exit with the return value
.I nr.
.br
The integers
.I err_nr_warnings
and 
.I err_nr_errors
hold the number of time these routines are called.
.br
The
.I err_*_jmpbuf
variables are pointers to a 
.I jmp_buf.
When these are not NULL they should point to a valid
.I jmp_buf
for use by
.I longjmp(3).
In this way the program can catch errors as generated by these routines.
.SH ARGUMENTS
The string
.I fmt
and trailing arguments are handled in the way of
.I printf(3).
.SH SEE ALSO
setjmp(3), printf(3), fprintf(3), exit(3).
.SH AUTHOR
Klamer Schutte, klamer@mi.el.utwente.nl


debian/changelog
clippoly (0.11-9) UNRELEASED; urgency=medium

  [ Ondřej Nový ]
  * d/watch: Use https protocol

  [ Barak A. Pearlmutter ]
  * bump policy

 -- Barak A. Pearlmutter <bap@debian.org>  Sat, 26 Jan 2019 20:07:14 +0000

clippoly (0.11-8) unstable; urgency=medium

  * dh11
  * bump policy version
  * swizzle packaging repo to salsa

 -- Barak A. Pearlmutter <bap@debian.org>  Fri, 02 Nov 2018 14:29:20 +0000

clippoly (0.11-7) unstable; urgency=medium

  * bump policy version
  * dh10
  * build dependencies for autoconf macros
  * harden
  * secure repo url
  * merge upstream tweaks: spelling

 -- Barak A. Pearlmutter <bap@debian.org>  Mon, 27 Feb 2017 09:41:20 +0000

clippoly (0.11-6) unstable; urgency=medium

  * support pkg-config clippoly by including .pc file

 -- Barak A. Pearlmutter <bap@debian.org>  Mon, 19 May 2014 10:59:28 +0100

clippoly (0.11-5) unstable; urgency=medium

  * bump policy version
  * lintian vcs-field-not-canonical

 -- Barak A. Pearlmutter <bap@debian.org>  Wed, 09 Apr 2014 14:20:11 +0100

clippoly (0.11-4) unstable; urgency=low

  * flush debian/rules standard dh target phony declarations (closes: #720692)
  * support multiarch

 -- Barak A. Pearlmutter <bap@debian.org>  Sat, 24 Aug 2013 22:47:51 +0100

clippoly (0.11-3) unstable; urgency=low

  * bump policy version
  * insert m4 doorstop
  * dh9
  * multilib support
  * change priority from extra to optional
  * create and count rely upon m4/doorstop

 -- Barak A. Pearlmutter <bap@debian.org>  Mon, 16 Apr 2012 20:07:40 +0100

clippoly (0.11-2) unstable; urgency=low

  * fix typo in debian/control
  * work around quilt patch representation restriction issue

 -- Barak A. Pearlmutter <bap@debian.org>  Thu, 18 Aug 2011 09:27:24 +0200

clippoly (0.11-1) unstable; urgency=low

  * Initial release (Closes: #638013)

 -- Barak A. Pearlmutter <bap@debian.org>  Wed, 29 Jun 2011 15:21:36 +0200


MatLab.MEX/unusedCo.de/err.c
/*
 *    tutvis library

 *    Copyright (C) 1993  University of Twente

 *    klamer@mi.el.utwente.nl

 *    This library is free software; you can redistribute it and/or
 *    modify it under the terms of the GNU Library General Public
 *    License as published by the Free Software Foundation; either
 *    version 2 of the License, or (at your option) any later version.

 *    This library is distributed in the hope that it will be useful,
 *    but WITHOUT ANY WARRANTY; without even the implied warranty of
 *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 *    Library General Public License for more details.

 *    You should have received a copy of the GNU Library General Public
 *    License along with this library; if not, write to the Free
 *    Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */

/*
 *	error.c				Klamer Schutte 25/7/90
 *
 *	handle errors in a consistent way.
 */

#ifndef err_c  /* gsl 10/15/96 */
#define err_c 1
#endif

#include	<stdio.h>
#ifdef mac_h
#include    <stdarg.h>
#include    "err.h"
#else
#include	<varargs.h>
#endif
#include	<setjmp.h>

/* WARNING: setjmp is never called anywhere in clippoly code as far as I can tell.
            => longjmp's are non-functional and will probably crash code, if they are
            executed. gsl 10/17/96
 */

int	err_nr_errors=0, err_nr_warnings=0; /*added init. to 0. gsl 10/15/96*/
/* what is the type of a pointer to a jmp_buf? On sun4 int * will do */
#ifdef mac_h
long **err_warning_jmpbuf, **err_error_jmpbuf, **err_fatal_jmpbuf;
#else
int	*err_warning_jmpbuf, *err_error_jmpbuf, *err_fatal_jmpbuf;
#endif

/*VARARGS1*/
#ifdef mac_h
void warning( const char *fmt, ... ) /* added void return. gsl 10/17/96 */
 {
	va_list	args;

	va_start( args, fmt );
	vfprintf(stderr, fmt, args );
	va_end(args );

	err_nr_warnings++;

	if (err_warning_jmpbuf != NULL)
		longjmp( err_warning_jmpbuf, 1 );
}	
#else
void warning( fmt, va_alist )
char	*fmt;
va_dcl
{
	va_list	args;

	va_start( args );
	vfprintf(stderr, fmt, args );
	va_end(args );

	err_nr_warnings++;

	if (err_warning_jmpbuf != NULL)
		longjmp( err_warning_jmpbuf, 1 );
}
#endif	

/*VARARGS1*/
#ifdef mac_h
void error( const char *fmt, ... ) /* added void return. gsl 10/17/96 */
{
	va_list	args;

	va_start( args, fmt );
	vfprintf(stderr, fmt, args );
	va_end(args );

	err_nr_errors++;

	if (err_error_jmpbuf != NULL)
		longjmp( err_error_jmpbuf, 1 );
}	
#else
void error( fmt, va_alist )
char	*fmt;
va_dcl
{
	va_list	args;

	va_start( args );
	vfprintf(stderr, fmt, args );
	va_end(args );

	err_nr_errors++;

	if (err_error_jmpbuf != NULL)
		longjmp( err_error_jmpbuf, 1 );
}	
#endif

/*VARARGS2*/
#ifdef mac_h
void fatal_exit_nr( int nr, const char *fmt, ... ) /* added void return. gsl 10/17/96 */
{
	va_list	args;

	va_start( args, fmt );
	vfprintf(stderr, fmt, args );
	va_end(args );

	if (err_fatal_jmpbuf != NULL)
		longjmp( err_fatal_jmpbuf, 1 );

	exit( nr );
}	
#else
void fatal_exit_nr( nr, fmt, va_alist )
int	nr;
char	*fmt;
va_dcl
{
	va_list	args;

	va_start( args );
	vfprintf(stderr, fmt, args );
	va_end(args );

	if (err_fatal_jmpbuf != NULL)
		longjmp( err_fatal_jmpbuf, 1 );

	exit( nr );
}	
#endif

/*VARARGS1*/
#ifdef mac_h
void fatal( const char *fmt, ... ) /* added void return. gsl 10/17/96 */
{
	va_list	args;

	va_start( args, fmt );
	vfprintf(stderr, fmt, args );
	va_end(args );

	if (err_fatal_jmpbuf != NULL)
		longjmp( err_fatal_jmpbuf, 1 );

	exit( 1 );
}	
#else
void fatal( fmt, va_alist )
char	*fmt;
va_dcl
{
	va_list	args;

	va_start( args );
	vfprintf(stderr, fmt, args );
	va_end(args );

	if (err_fatal_jmpbuf != NULL)
		longjmp( err_fatal_jmpbuf, 1 );

	exit( 1 );
}	
#endif

#undef err_c
